crates/re_viewer/src/misc/tensor_to_gpu.rs

```diff
@@ -262,63 +262,138 @@ fn general_texture_creation_desc_from_tensor<'a>(
+                TensorData::I16(buf) => (cast_slice_to_cow(buf), TextureFormat::Rg16Sint),
+                TensorData::I32(buf) => (cast_slice_to_cow(buf), TextureFormat::Rg32Sint),
+                TensorData::I64(buf) => (narrow_i64_to_f32s(buf), TextureFormat::Rg32Float), // narrowing to f32!
+
+                // TensorData::F16(buf) => (cast_slice_to_cow(buf), TextureFormat::Rg16Float), TODO(#854)
+                TensorData::F32(buf) => (cast_slice_to_cow(buf), TextureFormat::Rg32Float),
+                TensorData::F64(buf) => (narrow_f64_to_f32s(buf), TextureFormat::Rg32Float), // narrowing to f32!
+
+                TensorData::JPEG(_) => {
+                    anyhow::bail!("JPEGs should have been decoded at this point")
+                }
+            }
+        }
+        3 => {
+            // There are no 3-channel textures in wgpu, so we need to pad to 4 channels.
+            // What should we pad with? It depends on whether or not the shader interprets these as alpha.
+            // To be safe, we pad with the MAX value of integers, and with 1.0 for floats.
+            // TODO(emilk): tell the shader to ignore the alpha channel instead!
+            match &tensor.data {
+                TensorData::U8(buf) => (
+                    pad_and_cast(buf.as_slice(), u8::MAX),
+                    TextureFormat::Rgba8Uint,
+                ),
+                TensorData::U16(buf) => (pad_and_cast(buf, u16::MAX), TextureFormat::Rgba16Uint),
+                TensorData::U32(buf) => (pad_and_cast(buf, u32::MAX), TextureFormat::Rgba32Uint),
+                TensorData::U64(buf) => {
+                    // narrowing to f32!
+                    let pad = u64::MAX as f32;
+                    let floats: Vec<f32> = buf
+                        .chunks_exact(3)
+                        .flat_map(|chunk| [chunk[0] as f32, chunk[1] as f32, chunk[2] as f32, pad])
+                        .collect();
+                    (
+                        pod_collect_to_vec(&floats).into(),
+                        TextureFormat::Rgba32Float,
+                    )
+                }
+
+                TensorData::I8(buf) => (pad_and_cast(buf, i8::MAX), TextureFormat::Rgba8Sint),
+                TensorData::I16(buf) => (pad_and_cast(buf, i16::MAX), TextureFormat::Rgba16Sint),
+                TensorData::I32(buf) => (pad_and_cast(buf, i32::MAX), TextureFormat::Rgba32Sint),
+                TensorData::I64(buf) => {
+                    // narrowing to f32!
+                    let pad = i64::MAX as f32;
+                    let floats: Vec<f32> = buf
+                        .chunks_exact(3)
+                        .flat_map(|chunk| [chunk[0] as f32, chunk[1] as f32, chunk[2] as f32, pad])
+                        .collect();
+                    (
+                        pod_collect_to_vec(&floats).into(),
+                        TextureFormat::Rgba32Float,
+                    )
```

Comment:
you had all these nice helper functions making these large match expressions bearable, and now you stopped 😢